lib/curly/presenter.rb

```diff
@@ -285,18 +288,21 @@ def compute_cache_key
         [name, version, dependency_cache_keys].flatten.join("/")
       end
 
-      def presents(*args)
-        options = args.extract_options!
+      def presents(*args, **options, &block)
+        if options.key?(:default) && block_given?
+          raise ArgumentError,  "Cannot provide both `default:` and block"
+        end
 
         self.presented_names += args.map(&:to_s)
 
-        if options.key?(:default)
-          default_values = args.each_with_object(Hash.new) do |arg, hash|
-            hash[arg.to_s] = options.fetch(:default)
+        args.each do |arg|
+          self.default_values = self.default_values.
+          merge arg.to_s => if options.key?(:default)
+            [:value, options[:default]]
+          else
+            [:block, block]
```

Comment:
This could be a bit cleaner:

```ruby
self.default_values = self.default_values.
  merge(arg.to_s => options.key?(:default) ? [:value, options[:default]] : [:block, block])
```